skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,178 @@
+            f"was {type(intensity_vertex)}")
+
+    if space_vertex <= 0:
+        raise ValueError(
+            f"Spacial radius must be greater zeros, was {space_vertex}")
+
+    if intensity_vertex <= 0:
+        raise ValueError(
+            f"Intensity radius must be greater zeros, was {intensity_vertex}")
+
+    try:
+        white_background = bool(white_background)
+    except ValueError:
+        raise ValueError(
+            f"white_background must be convertible to a boolean, "
+            f"was {type(white_background)}"
+        )
+
+    if not isinstance(image, np.ndarray):
+        raise ValueError(
+            f"input_img must be a np.ndarray, "
+            f"was {type(image)}"
+        )
+
+    if not issubclass(image.dtype.type, np.integer):
+        raise ValueError("Currently only integer images are supported.")
+
+    img = image.copy()
+    if white_background:
+        img = invert(img)
+
+    # assuming selem touches the umbra at (x,y) pre-compute the
+    # (relative) height of selem at the center
+    # (selem is a ball/spheroid)
+    L = np.arange(-space_vertex, space_vertex + 1)
+    X, Y = np.meshgrid(L, L)
+    distance = np.sqrt(X ** 2 + Y ** 2)
+    sagitta = space_vertex - space_vertex * np.sqrt(
+        np.clip(1 - (distance ** 2 / intensity_vertex ** 2), 0, None)
+    )
+
+    kernel = np.array(distance <= space_vertex, dtype=float)
+    kernel[kernel == 0] = np.Inf
+
+    kernel_size = np.array(kernel.shape)
+    img_original = img.copy()
+    img_size = img.shape
+    x = img
+    stride = (1, 1)
+
+    # pad the image
+    padding_amount = kernel_size // 2 * 2
+    half_pad = padding_amount // 2
+    img = np.Inf * np.ones(x.shape + padding_amount, dtype=x.dtype)
+    img[half_pad[0]:-half_pad[0], half_pad[1]:-half_pad[1]] = x
+    large_img_size = img.shape
+
+    # the following three blocks implement a variant of 2d convolution
+    # in python. Each time a kernel is applied to a window, sagitta is added
+    # to the respective pixel values and the reduction is done using `min` instead
+    # of `sum`
+
+    # window of affected pixel indices relative to anchor
+    x_idx = np.arange(kernel_size[1])
+    y_idx = large_img_size[1] * np.arange(kernel_size[0])
+    kernel_idx = (x_idx[np.newaxis, :] + y_idx[:, np.newaxis]).flatten()
+
+    # indices of each ancor for a kernel
+    # (top left corner instead of center)
+    x_anchors = np.arange(
+        large_img_size[1] - kernel_size[1] + 1, step=stride[1])
+    y_anchors = large_img_size[1] * \
+        np.arange(large_img_size[0] - kernel_size[0] + 1, step=stride[0])
+    anchor_offsets = (x_anchors[np.newaxis, :] +
+                      y_anchors[:, np.newaxis]).flatten()
+
+    # compute px indices in the image and apply the function/kernel
+    # large images or kernel sizes don't fit into memory
+    # do it in batches instead
+    background = np.zeros(img_original.size)
+    batch_size = int(2 ** 10)
+    flat_img = img.flatten()
```

Comment:
I think that you can use `img.reshape(-1)` (see the notes section in `np.ravel` [doc](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html#numpy.ravel)) to avoid the copy introduced by `np.flatten`

Reply:
great suggestion! I only recently became more conscious about numpy views and their utility. I'll check if I can apply this to other parts of the code, too.